File: 2446_PrintStar9/cpp/_2446.cpp
#include <iostream>


int main()
{
    int n;
    std::cin >> n;

    bool isIncreasing = false;
    int starNum = 2*n - 1;
    for(int i = 0 ; i < 2*n - 1 ; i++)
    {
        for(int j = 0; j < 2*n - 1; j++)
        {
            int noStarNum = ((2*n - 1) - starNum) / 2;
            if(j < noStarNum)
                std::cout << ' ';
            else if(j >= noStarNum + starNum)
                break;
            else
            {
                std::cout << '*';
            }
            
        }
        std::cout << std::endl;
        
        isIncreasing ? (starNum += 2) : (starNum -= 2);
        if(starNum < 0) {
            starNum = 3;
            isIncreasing = true;
        }
    }


    std::cin.get();
    return 0;
}

File: Baekjoon_CSharp/Baekjoon_CSharp/_2446.cs
using System;

namespace cs
{
    class Program
    {
        static void Main(string[] args)
        {
            int n;
            n = int.Parse(Console.ReadLine()); 

            int starNum = 2*n - 1;
            int noStarNum = 0;
            bool isIncreasing = false;
            for(int i = 0 ; i < 2*n - 1 ; i++)
            {
                for(int j = 0 ; j < starNum + noStarNum; j++)
                {
                    if(j < noStarNum)
                        Console.Write(' ');
                    else 
                        Console.Write('*');
                }
                Console.Write('\n');
                if(starNum == 1) isIncreasing = true;

                if(!isIncreasing){
                    noStarNum += 1;
                    starNum -= 2;
                }
                else{
                    noStarNum -= 1;
                    starNum += 2;
                }
            }

            Console.Read();
        }
    }
}
